### ISSUE TITLE
Get variations by sku

### ISSUE THREAD

**Issue by maxim-skuvault** [woocommerce/woocommerce-rest-api#38]
It would be very useful to have an endpoint to get variations by sku, without having so supply the productId.

We use the sku to look up variations in WooCommerce and then update the stock quantity. Currently, to get a variation from the api, the only option is to pass the parent productId, which we don't have nor do we want to keep track of outside wooCommerce. This severely limits the usefulness of the api for the purpose of updating quantities. We're having to get **all** products and then all variations for each product and then filter variations by sku, which is a huge performance hit

**Comment by vedanshujain** [comment]
This is supported by using `sku` param to filter the product list. For egs, if you'd like to get a product with sku `test_123` the request can be written like so:
`/wp-json/wc/v3/products?sku=pro-123`

Similarly `sku` filter is also supported for product variations as well. However, you would still need to do at least one request with `sku` filter to get product id, then another to get to the variations. But this should be lot faster than getting all the products and then parsing through to get to the variation. 

That said, I am closing the issue because we likely won't support endpoints with `sku` embedded instead of `id` since the above workaround is probably good enough for the majority of uses. But feel free to write back if that's not the case and this can be discussed further.

**Comment by yordivd** [comment]
@vedanshujain I really don't understand your answer.
We have 1000's products with each +-6 variations.
Every variation has an unique sku which is printed on a barcode.

E.g.: 
Product sku: 5876
Product variations sku: 26298, 16156, 16616

When you make a POS and you want to scan the barcode you can scan this sku code to retrieve the product/variation.
So i can the product variation with sku 16156, how on earth can my POS efficiently get this product variation? Since it always needs the parent sku to search variations, but it doesn't know this when scanning the barcode.

I can not wrap my head around why the woocommerce dev team does not see the issue in this. Or am i missing something here?
You are basically saying: if you need to search variations you should sync all products to your own local database to be able to work with it/search in it.

**Comment by jorgemd24** [comment]
You can use /wp-json/wc/v3/products?sku=variation_sku  to get the variation id and the parent_id (product_id)

**Comment by diickens** [comment]
@jorgemd24 When I try to get the variation with this endpoint, I get a critical Error response from the server.
When I try this endpoint for the parent product it works perfect. Any Idea?

**Comment by maxim-skuvault** [comment]
@jorgemd24, You posted:
> You can use /wp-json/wc/v3/products?sku=variation_sku to get the variation id and the parent_id (product_id)

Has this worked for you? When I created this issue the /wp-json/wc/v3/products endpoint would only search by the product sku, not variation sku (what I need). The [documentation](https://woocommerce.github.io/woocommerce-rest-api-docs/#list-all-products) still implies that it's only by product sku: "sku: Limit result set to products with a specific SKU"

**Comment by jorgemd24** [comment]
@maxim-skuvault  

> Has this worked for you?

Yes it does work, have you tried it recently?

> sku: Limit result set to products with a specific SKU

I think they are talking "in the broad sense" of products (parent products, variations, simple products, groups etc). Otherwise they will need to specify that only works for parent products or simple products.

**Comment by jorgemd24** [comment]
@diickens 
>  I get a critical Error response from the server.

Do you have more information about the error? have you tried to debug it?
